core/stats.go

```diff
@@ -57,24 +69,60 @@ func (ds *dockerService) ListContainerStats(
 		return nil, err
 	}
 
-	var mtx sync.Mutex
-	var wg sync.WaitGroup
-	var stats = make([]*runtimeapi.ContainerStats, 0, len(listResp.Containers))
-	for _, container := range listResp.Containers {
-		container := container
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			if containerStats, err := ds.getContainerStats(container.Id); err == nil && containerStats != nil {
-				mtx.Lock()
-				stats = append(stats, containerStats)
-				mtx.Unlock()
-			} else if err != nil {
-				logrus.Error(err, " Failed to get stats from container "+container.Id)
+	g, ctx := errgroup.WithContext(ctx)
```

Comment:
Can I have you add some inline comments to make it easier to remember what this code is doing the next time around

Reply:
I just added some comments in a new commit, please let me know if it's OK.